homeassistant/components/zwave_js/climate.py

```diff
@@ -0,0 +1,332 @@
+        return [
+            self.get_zwave_value(
+                ZWAVE_SETPOINT_PROPERTY,
+                CommandClass.THERMOSTAT_SETPOINT,
+                value_property_key_name=key,
+                add_to_watched_value_ids=True,
+            )
+            for key in setpoint_keys
+            if self.get_zwave_value(
+                ZWAVE_SETPOINT_PROPERTY,
+                CommandClass.THERMOSTAT_SETPOINT,
+                value_property_key_name=key,
+                add_to_watched_value_ids=True,
+            )
+        ]
+
+    def _set_modes_and_presets(self) -> None:
+        """Convert Z-Wave Thermostat modes into Home Assistant modes and presets."""
+        all_modes: Dict[str, Optional[int]] = {}
+        all_presets: Dict[str, Optional[int]] = {PRESET_NONE: None}
+        if self.info.primary_value:
+            # Z-Wave uses one list for both modes and presets.
+            # Iterate over all Z-Wave ThermostatModes and extract the hvac modes and presets.
+            for mode_id, mode_name in self.info.primary_value.metadata.states.items():
+                mode_id = int(mode_id)
+                if mode_id in THERMOSTAT_MODES:
+                    # treat value as hvac mode
+                    hass_mode = ZW_HVAC_MODE_MAP.get(mode_id)
+                    if hass_mode:
+                        all_modes[hass_mode] = mode_id
+                else:
+                    # treat value as hvac preset
+                    all_presets[mode_name] = mode_id
+        else:
+            all_modes[HVAC_MODE_HEAT] = None
+        self._hvac_modes = all_modes
+        self._hvac_presets = all_presets
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        temp: ZwaveValue = (
+            self._current_mode_setpoint_values[0]
+            if self._current_mode_setpoint_values
+            else None
+        )
+        if temp is not None and temp.metadata.unit == "°F":
+            return TEMP_FAHRENHEIT
+        return TEMP_CELSIUS
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        if not self.info.primary_value:
+            # Thermostat(valve) with no support for setting a mode is considered heating-only
+            return HVAC_MODE_HEAT
+        return ZW_HVAC_MODE_MAP.get(
+            int(self.info.primary_value.value), HVAC_MODE_HEAT_COOL
+        )
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes."""
+        return list(self._hvac_modes)
+
+    @property
+    def hvac_action(self) -> Optional[str]:
+        """Return the current running hvac operation if supported."""
+        operating_state = self.get_zwave_value(
+            "state",
+            command_class=CommandClass.THERMOSTAT_OPERATING_STATE,
+            add_to_watched_value_ids=True,
```

Comment:
It would be good if we didn't have to add values to watched values after set up. Can we make sure the set up is complete when the node is ready and the entity has been added?

Reply:
Maybe I misunderstood the purpose of adding to watched values - I thought the point here was to monitor changes to that value and to write a state update? The default is also to add to watched values so I assumed in most cases we want  this behavior.

Reply:
Yes, and our interest in a value shouldn't change for the entity after set up.

Reply:
OK so I should add the values during initialization, and that way it will be set up by the time the node is ready and entity has been added.

Good thing there's more todo for the release, I think I will need a couple of additional constants

Reply:
how can I act on a node is ready event? If I am understanding things correctly, the order of operations can get mixed (initialization, entity added, node ready), so I need to be able to add additional values to watch only after all three are true

Reply:
nvm found the event

Reply:
No, we'll only set up entities for nodes that are ready.